lightning/src/ln/outbound_payment.rs

```diff
@@ -717,7 +724,8 @@ impl OutboundPayments {
 	#[allow(unused)]
 	pub(super) fn send_payment_for_bolt12_invoice<R: Deref, ES: Deref, NS: Deref, IH, SP, L: Deref>(
 		&self, invoice: &Bolt12Invoice, payment_id: PaymentId, router: &R,
-		first_hops: Vec<ChannelDetails>, inflight_htlcs: IH, entropy_source: &ES, node_signer: &NS,
+		first_hops: Vec<ChannelDetails>, max_total_routing_fee_msat: Option<u64>,
```

Comment:
This will be called from `ChannelManager` for the initial payment (see #2039). Retries hit the normal code path. If we want users to be able to set this, it needs to piped from `ChannelManager::pay_for_offer` (see #2578) into `OutboundPayments::add_new_awaiting_invoice` (already merged) so that it can be included in `PendingOutboundPayment::AwaitingInvoice` and used here.